### PR TITLE
Improve build docs discoverability and add some missing macro documentation (like ASIO_NO_DEPRECATED )

### PR DESCRIPTION
**Pull request by audaki**

### CODE REVIEW

File: asio/src/doc/using.qbk [hunk 1]
```diff
@@ -292,6 +292,48 @@ Asio.
...
+  [
+    [`ASIO_DISABLE_MOVE`]
+    [
+      Explicitly disables move support for all compilers.
+    ]
+  ]
+  [
+    [`ASIO_NO_DEPRECATED`]
+    [
+      Enables compile-time detection of the deprecated `asio_handler_invoke`
```
Comment (mabrarov) [inline asio/src/doc/using.qbk:314]: What about `asio_handler_allocate` and `asio_handler_deallocate` (e.g. [asio/detail/wrapped_handler.hpp](https://github.com/chriskohlhoff/asio/blob/master/asio/include/asio/detail/wrapped_handler.hpp#L208))?

Comment (audaki) [reply asio/src/doc/using.qbk]: @mabrarov You're right there were 2 bullet points in the changelog. I've merged them into one explanation for the macro.

**Comment by audaki** [comment]
Heya, is there anything I can still do for this pull request? :-)

**Review by TheStormN — APPROVED** [review]

**Comment by audaki** [comment]
@mabrarov Maybe it helps getting this merged to change your review status from "left comments" to "approved"? ;-)

**Review by mabrarov — APPROVED** [review]
Looks good. I appreciate if this pull request could be merged soon.

**Comment by audaki** [comment]
@chriskohlhoff Heya, is there anything we can do to improve this pull request?

**Comment by chriskohlhoff** [comment]
In asio 1.18.1 / boost 1.75 I applied some of these changes, but took a different approach for the others.

**Pull request closed without merging**